src/middlewares/auth.middleware.ts

```diff
@@ -0,0 +1,21 @@
+import { NextFunction, Request, Response } from 'express';
+import { AuthService } from '../services';
+import { InternalError, ErrorType } from '../models';
+
+const authMiddleware = (service: AuthService) => (
+  request: Request,
+  response: Response,
+  next: NextFunction
+): Promise<void> => {
+  const authHeader = request.headers.authorization;
+  console.log('header', authHeader);
+  if (!authHeader) {
+    return Promise.resolve(next(new InternalError('Authorization failed', ErrorType.UNAUTHORIZED)));
+  }
+  return service
+    .validateToken(authHeader)
```

Comment:
authHeader should be in format: 'Bearer <token>'

Reply:
done